lib/backends/sdc/containers.js

```diff
@@ -681,43 +667,96 @@ function defaultFabricNetwork(opts, container, payload, callback) {
+    // 3. partial id match
     vasync.parallel({
         funcs: [
+            function byExactId(cb) {
+                // length === 64, and 0..31 === 32..63, or it's an 'impossible'
+                // id under our double-uuid convention, and we can skip it.
+                if (query.substr(0, 32) !== query.substr(32)) {
+                    log.debug({ query: query },
+                        'Networks: impossible exactId: %s, skipping', query);
+                    setImmediate(cb, null, []);
+                    return;
+                }
+                var uuid = utils.shortNetworkIdToUuid(query);
+
+                // XXX - ldapEscape required to work around NAPI-367
+                var listParams = {
+                    uuid: utils.ldapEscape(uuid),
+                    fabric: true,
+                    owner_uuid: opts.account.uuid
+                };
+
+                napi.listNetworks(listParams,
+                    { headers: { 'x-request-id': opts.req_id }}, cb);
+            },
             function byName(cb) {
                 var listParams = {
                     name: query,
                     fabric: true,
-                    owner_uuid: opts.account.uuid,
-                    limit: 2
+                    owner_uuid: opts.account.uuid
                 };
 
                 log.debug({ listParams: listParams },
                     format('Networks: searching for network %s',
                         listParams.name));
 
-
                 napi.listNetworks(listParams,
                     { headers: {'x-request-id': opts.req_id }}, cb);
             },
             function byDockerId(cb) {
-                // to search for docker ids, when NAPI only retains UUIDs, we
-                // rely on the 'double uuid' convention (see
-                // utils.networkUuidToDockerId()). To perform the search,
-                // we truncate the provided string to 8 chars, perform a
-                // wildcard search on networks, and post-filter the results to
-                // find more-complete matches than the at-most-8-char prefix
-                // might find.
+                // we assume the 'double uuid' convention for networks here,
+                // that is, dockerId = (uuid + uuid).replace(/-/g, '').
+                // So if we have a query.length > 31, 32... must be a prefix of
+                // 0..31. If not, the id supplied is impossible in our system.
+                if (query.length > 31
+                    && query.substr(0, query.substr(32).length)
+                    === query.substr(32)) {
+
+                    log.info({ query: query },
+                        'Networks: impossible network id query %s, skipping');
+                    setImmediate(cb, null, []);
+                    return;
+                }
 
-                var queryString = utils.ldapEscape(query.substr(0, 8)) + '*';
+                // To perform the search, we transform the provided query to
```

Comment:
Would it help readers to mention that the query itself can be a partial Docker ID, not only the result of the transformation from a Docker ID to a UUID?